app/shared/models/git_repo.rb

```diff
@@ -87,6 +88,21 @@ def pushes_disabled?
       def load_octokit_cache_stack
         @stack ||= Faraday::RackBuilder.new do |builder|
           builder.use(Faraday::HttpCache, serializer: Marshal, shared_cache: false)
+          # In order to have access to SocketError and ETIMEDOUT exceptions.
+          require "socket"
+          require "net/http"
+          builder.use(Faraday::Request::Retry,
```

Comment:
Instead of applying that logic here, I think we should improve the retry logic here: https://github.com/fastlane/ci/blob/master/app/shared/github_handler.rb#L36

That way we don't have multiple layers doing their own retry/backoff logic. 
It's better that we have custom control over how this works in 1 single place.